common/cache/lru.go

```diff
@@ -290,6 +302,7 @@ func (c *lru) putInternal(key interface{}, value interface{}, allowUpdate bool)
 	entry := &entryImpl{
 		key:   key,
 		value: value,
+		size:  getSize(value),
```

Comment:
shall we still do the check of size v.s. cache max size and return error early if item size is too big.

Reply:
sure, I move the size validation on the top before entry created.

Reply:
I guess I mean we don't have to evict any entry if new item size is too big.

Reply:
I see, thanks for clarify. I don't feel strong here, I will make the change to do the check before.

Reply:
actually, run the loop twice will make the logic looks duplicate and complex. do you feel strong about check before eviction?